auto-farm/src/fees_collector_actions.rs

```diff
@@ -0,0 +1,54 @@
+use common_structs::PaymentsVec;
+
+use crate::user_rewards::UniquePayments;
+
+elrond_wasm::imports!();
+
+#[elrond_wasm::module]
+pub trait FeesCollectorActionsModule:
+    crate::common_storage::CommonStorageModule
+    + crate::user_rewards::UserRewardsModule
+    + crate::fees::FeesModule
+    + utils::UtilsModule
+    + energy_query::EnergyQueryModule
+    + crate::locked_token_merging::LockedTokenMergingModule
+    + lkmex_transfer::energy_transfer::EnergyTransferModule
+    + legacy_token_decode_module::LegacyTokenDecodeModule
+{
+    #[endpoint(claimFeesCollectorRewards)]
+    fn claim_fees_collector_rewards(&self, user: ManagedAddress) {
+        self.require_caller_proxy_claim_address();
+
+        let mut rewards = self.call_fees_collector_claim(user.clone());
+        let rewards_len = rewards.len();
+        if rewards_len == 0 {
+            return;
+        }
+
+        // locked token rewards, if any, are always in the last position
```

Comment:
Not sure if it is safe to always assume this. In future, in case of computation changes in the fees collector sc, this logic would become obsolete. Which in turn means the devs should know that they need to update this function as well, when updating that one.